pkg/olm/convert/registryv1.go

```diff
@@ -0,0 +1,509 @@
+		for {
+			obj := unstructured.Unstructured{}
+			err := dec.Decode(&obj)
+			if errors.Is(err, io.EOF) {
+				break
+			}
+			if err != nil {
+				return nil, fmt.Errorf("read %q: %v", e.Name(), err)
+			}
+			objects = append(objects, &obj)
+		}
+	}
+
+	for _, obj := range objects {
+		obj := obj
+		switch obj.GetObjectKind().GroupVersionKind().Kind {
+		case "ClusterServiceVersion":
+			csv := v1alpha1.ClusterServiceVersion{}
+			if err := runtime.DefaultUnstructuredConverter.FromUnstructured(obj.Object, &csv); err != nil {
+				return nil, err
+			}
+			reg.CSV = csv
+		case "CustomResourceDefinition":
+			crd := apiextensionsv1.CustomResourceDefinition{}
+			if err := runtime.DefaultUnstructuredConverter.FromUnstructured(obj.Object, &crd); err != nil {
+				return nil, err
+			}
+			reg.CRDs = append(reg.CRDs, crd)
+		default:
+			reg.Others = append(reg.Others, *obj)
+		}
+	}
+
+	plain, err := Simple(reg)
+	if err != nil {
+		return nil, err
+	}
+
+	var manifest bytes.Buffer
+	for _, obj := range plain.Objects {
+		yamlData, err := yaml.Marshal(obj)
+		if err != nil {
+			return nil, err
+		}
+		if _, err := fmt.Fprintf(&manifest, "---\n%s\n", string(yamlData)); err != nil {
+			return nil, err
+		}
+	}
+
+	now := time.Now()
+	plainFS := fstest.MapFS{
+		".": &fstest.MapFile{
+			Data:    nil,
+			Mode:    fs.ModeDir | 0755,
+			ModTime: now,
+		},
+		"manifests": &fstest.MapFile{
+			Data:    nil,
+			Mode:    fs.ModeDir | 0755,
+			ModTime: now,
+		},
+		"manifests/manifest.yaml": &fstest.MapFile{
+			Data:    manifest.Bytes(),
+			Mode:    0644,
+			ModTime: now,
+		},
+	}
+
+	return plainFS, nil
+}
+
+func validateTargetNamespaces(supportedInstallModes sets.Set[string], installNamespace string, targetNamespaces []string) error {
+	set := sets.New[string](targetNamespaces...)
+	switch set.Len() {
+	case 0:
+		if supportedInstallModes.Has(string(v1alpha1.InstallModeTypeAllNamespaces)) {
+			return nil
+		}
+	case 1:
+		if set.Has("") && supportedInstallModes.Has(string(v1alpha1.InstallModeTypeAllNamespaces)) {
+			return nil
+		}
+		if supportedInstallModes.Has(string(v1alpha1.InstallModeTypeSingleNamespace)) {
+			return nil
+		}
+		if supportedInstallModes.Has(string(v1alpha1.InstallModeTypeOwnNamespace)) && targetNamespaces[0] == installNamespace {
+			return nil
+		}
+	default:
+		if supportedInstallModes.Has(string(v1alpha1.InstallModeTypeMultiNamespace)) {
+			return nil
+		}
+	}
+	return fmt.Errorf("supported install modes %v do not support target namespaces %v", sets.List[string](supportedInstallModes), targetNamespaces)
+}
+
+func Simple(in RegistryV1) (*Plain, error) {
+	return Convert(in, "", nil)
+}
+
+func saNameOrDefault(saName string) string {
+	if saName == "" {
+		return "default"
+	}
+	return saName
+}
+
+func Convert(in RegistryV1, installNamespace string, targetNamespaces []string) (*Plain, error) {
+	if installNamespace == "" {
+		installNamespace = in.CSV.Annotations["operatorframework.io/suggested-namespace"]
```

Comment:
Nit for now, but functional change: We need to use the installNamespace from `AppSpec.DefaultNamespace`, instead of defaulting to `<packageName>-system`. (This would kind of bring us to a behaviour similar to OLM v0, where the install namespace is same as where the subscription is created?).